server/lib/device/device.getDeviceFeaturesStates.js

```diff
@@ -0,0 +1,73 @@
+const { Op } = require('sequelize');
+const db = require('../../models');
+const { NotFoundError } = require('../../utils/coreErrors');
+
+const DEFAULT_OPTIONS = {
+  skip: 0,
+  order_dir: 'ASC',
+  order_by: 'created_at',
+};
+
+/**
+ * @description Get all features states aggregates.
+ * @param {string} selector - Device selector.
+ * @param {Object} [options] - Options of the query.
+ * @param {string} [options.from] - Start date in UTC format "yyyy-mm-ddThh:mm:ss:sssZ"
+ * or "yyyy-mm-dd hh:mm:ss:sss" (GMT time).
+ * @param {string} [options.to] - End date in UTC format "yyyy-mm-ddThh:mm:ss:sssZ"
+ * or "yyyy-mm-dd hh:mm:ss:sss" (GMT time).
+ * @param {number} [options.take] - Number of elements to return.
+ * @param {number} [options.skip] - Number of elements to skip.
+ * @param {string} [options.attributes] - Possible values (separated by a comma ',' if several): 'id',
+ * 'device_feature_id', 'value', 'created_at' and 'updated_at'. Leave empty to have all the columns.
+ * @returns {Promise<Object>} - Resolve with an array of data.
+ * @example
+ * device.getDeviceFeaturesStates('test-device', [from: '2022-03-31T00:00:00.000Z', to: '2022-03-31T23:59:59.999Z',
```

Comment:
example if not correct, options is not an array

Reply:
Sorry, I'm not sure I understood everything about this request, are you asking me here for 2 separate things or only the modification of the [] to {} for the option which is an object and not an array?
- [ ] Modified the brackets [...] by braces{...} to signify an object and not an array: 
- [ ] Add an example with a 400 bad request return? (FYI if it's about that, I didn't do it because I haven't seen any other example of failure in the current Rest API documentation. But I can add it if needed ^^)